config/routes.rb

```diff
@@ -1,8 +1,8 @@
-# frozen_string_literal: true
-
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root 'users#index'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users/:user_id/posts/:id', to: 'posts#index', as: 'user_post'
+  get '/users/:user_id/posts', to: 'posts#show', as: 'user_posts'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users', to: 'users#index', as: 'users'
```

Comment:
- [ ] When I run the tests for post, I get this error below. Please ensure you are testing with the right routes
<img width="712" alt="Screenshot 2023-10-03 at 12 04 21" src="https://github.com/abdozayan12/DataApp/assets/64774936/571d5b10-6969-4ebb-847b-b166a2c136fa">